portal-web/test/functional/com/liferay/portalweb/macros/SAML.macro

```diff
@@ -1,5 +1,125 @@
 definition {
 
+	@description = "Add new IDP Connection data for SP."
+	macro addNewIDPConnection {
+		Navigator.openSpecificURL(url = "${spURL}");
+
+		if (IsElementNotPresent(locator1 = "UserBar#USER_AVATAR_IMAGE")) {
+			if (isSet(spAdminEmailAddress)) {
+				User.loginPG(
+					password = "test",
+					specificURL = "${spURL}",
+					userEmailAddress = "${spAdminEmailAddress}");
+			}
+			else {
+				User.loginPG(
+					password = "test",
+					specificURL = "${spURL}",
+					userEmailAddress = "test@liferay.com");
+			}
+		}
```

Comment:
Login shouldn't be included in this macro since it's supposed to handle adding a new IDP connection. We want to make sure someone who's new to SAML can follow the steps from the testcase level.